node-src/git/getChangedFilesWithReplacement.ts

```diff
@@ -18,17 +19,18 @@ type BuildWithCommit = {
  */
 export async function getChangedFilesWithReplacement(
   context: Context,
-  build: BuildWithCommit
-): Promise<{ changedFiles: string[]; replacementBuild?: BuildWithCommit }> {
+  build: BuildWithCommitInfo
+): Promise<{ changedFiles: string[]; replacementBuild?: BuildWithCommitInfo }> {
   try {
+    if (build.uncommittedHash) throw new Error('Build had uncommitted changes');
     const changedFiles = await getChangedFiles(build.commit);
     return { changedFiles };
   } catch (err) {
     context.log.debug(
       `Got error fetching commit for #${build.number}(${build.commit}): ${err.message}`
     );
 
-    if (err.message.match(/bad object/)) {
+    if (err.message.match(/(bad object|uncommitted changes)/)) {
```

Comment:
Are we OK with throwing for flow control? 

I think maybe it would be less confusing to do:

```ts
let commitErr;
try {
  if (!build.uncommittedHash) {
     const changedFiles = await getChangedFiles(build.commit);
     return { changedFiles };
  }
} catch (err) {
   if (!err.message.match(/bad object/)) {
     throw err;
   }
   commitErr = err;
}

// now do replacement logic

// if fails
throw commitErr;
```

I'm not super fussed about it though.

Reply:
I would've done that if not for the preexisting catch logic. It seemed more sensible to extend that rather than pile on a 2nd kind of "exception" handling.